### PR TITLE
Guardian UI - Setup completion flow

### PR DESCRIPTION
**Pull request by wbobeirne**
### What This Does

* Implements a `SetupComplete` component for after you've run consensus
* Implements a placeholder `FederationDashboard` component that renders when your federation is running
  * When you visit the page after setup, you will always be directed to this page
  * We will need more APIs to actually make this page useful, but hopefully it will provide the federation join code and some basic status information going forward, maybe even administrative actions? #2451
* Fixes and improves `GuardianApi.startConsensus()`
  * Properly tears down and restarts socket, previously it would keep trying to use the busted socket
  * Retries up to 10 times with 1 second intervals to allow for slower restarts
* Returns the full `StatusResponse` from `GuardianApi.status()`
  * Updated all places in code we used this from `status` to `status.server`

## Screenshots

https://user-images.githubusercontent.com/649992/236887394-5c93f3f6-fd33-4b0a-b6da-1b4f59c23132.mp4

### CODE REVIEW

File: guardian-ui/src/components/SetupComplete.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      justify='center'
+      align='center'
+      textAlign='center'
+      pt={10}
+    >
+      <Heading size='sm' fontSize='42px' mb={8}>
+        🎉 🎉 🎉
+      </Heading>
+      <Heading size='md' fontWeight='medium' mb={2}>
+        Congratualations
```
Comment (garyray-k) [inline guardian-ui/src/components/SetupComplete.tsx:26]: nit: typo

File: guardian-ui/src/components/SetupComplete.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Flex, Heading, Text, Button, Icon } from '@chakra-ui/react';
+import React, { useCallback } from 'react';
+import { ReactComponent as ArrowRightIcon } from '../assets/svgs/arrow-right.svg';
+import { useGuardianContext } from '../hooks';
+import { SETUP_ACTION_TYPE } from '../types';
+
+export const SetupComplete: React.FC = () => {
+  const { dispatch } = useGuardianContext();
+
+  const handleContinue = useCallback(() => {
```
Comment (garyray-k) [inline guardian-ui/src/components/SetupComplete.tsx:10]: If we've made it this far, should we SET_IS_SETUP_COMPLETE on mount? Why wait for the button click?

File: guardian-ui/src/types.tsx [hunk 3]
```diff
@@ -77,10 +77,17 @@ export interface ConsensusState {
...
-    api: string;
+    consensus: number;
+    api: { major: number; minor: number }[];
   };
-  modules: Record<number, { core: string; module: string; api: string }>;
+  modules: Record<
+    number,
+    {
+      core: string;
+      module: string;
```
Comment (garyray-k) [inline guardian-ui/src/types.tsx:87]: Aren't `core` and `module` also numbers?

**Review by garyray-k — CHANGES_REQUESTED** [review]

**Comment by wbobeirne** [comment]
Thank you for the eagle-eyed review @garyray-k!

**Review by garyray-k — APPROVED** [review]

**Pull request merged**